File: javascript-30/day08.js
function processData(input) {
    //Enter your code here
    input = input.split('\n');
    var phonebook = [];
    var entry = parseInt(input[0]);

    for (var j = 1; j <= entry; j++){
        var contactArray = input[j].split(' ');
        phonebook[contactArray[0]] = contactArray[1];
    }

    for (var j = entry + 1; j < input.length; j++){
        var number = (phonebook[input[j]]);
        if (number !== undefined){
            console.log(input[j] +'='+ number);
        } else{
            console.log('Not found');
        }
    }
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: python-30/day08.py
# Enter your code here. Read input from STDIN. Print output to STDOUT

n = int(input())
# split the name and the number
name_num = [input().split() for _ in range(n)]
# the phone number dictionary
phone_dict = {k: v for k,v in name_num}
while True:
    try:
        name = str(input())
        if name in phone_dict:
            print("{}={}".format(name, phone_dict[name]))
        else:
            print("Not found")
    except:
        break
